README-dev.md

```diff
@@ -0,0 +1,106 @@
+* It depends on the [tsp-typescript-client][client] for server health check purposes.
+
+This extension was started from Code's [guide][guide] and related [sample][sample].
+
+## Documentation
+
+This README is the usual entry point for documenting this extension.
+
+* One may refer to the [contribution guide](CONTRIBUTING.md) for how to contribute.
+* Please also refer to the [security policy](SECURITY.md) on a need basis.
+* The usual [license description](LICENSE.md) file accompanies these too.
+
+## Build
+
+Run `yarn`, which should automatically include `yarn install`.
+
+* This extension is bundled using `webpack`, originally based on [the guide][guide].
+* There is only a modest automated CI test suite being run on GitHub
+
+## Test
+
+Run `yarn test` on a need basis.
+
+Alternatively, launch `Extension Tests` under `Run and Debug`.
+
+## Installation
+
+1. After [having built](#build) at least once, run `yarn vsce:package` ([more][vsce]) at will.
+1. [Install][install] the hereby generated `vscode-trace-server-*.vsix` file.
+1. Alternatively, simply launch the packaged extension using `Run Extension`.
+1. Through `Command Palette`, the `Trace Server:` start/stop commands should be available.
+
+This extension can be installed in either one (or many) of:
+
+* [VS Code][code] or [Codium][codium]/Code-OSS, or
+* a [Theia][theia] application such as [Blueprint][blueprint].
+
+The dependent [Trace Viewer for VSCode][vscode-trace-extension] extension renders a `Trace Server`
+[status bar item][item]. A note:
+
+Reinstalling an amended extension that has the same version requires removing the unpacked extension, found under one of the following folders:
+
+* [Theia Blueprint][blueprint]:  extracts installed extensions under `/tmp/vscode-unpacked/`.
+* VSCode: extracts the installed extensions under the user's home, in folder `.vscode/extensions/`.
+
+Alternatively, you may step the extension's version to avoid this issue.
+
+## Debugging
+
+* One may launch the extension using `Run Extension`, to debug it with breakpoints, as usual.
+* The same can be done for tests, launching `Extension Tests` to debug them.
+* The enabled breakpoints get bound only upon exercising the extension.
+
+## Development
+
+The usual [Prettier][prettier] and [ESLint][eslint] combo in VS Code or Codium OSS is used.
+
+* [This matcher][matcher] is also used, since the originally generated extension per [guide].
+* Markdown gets linted with the (usual) [vscode-markdownlint][markdownlint] extension.
+* [SonarLint][sonarlint] is also assumed while further developing this extension.
+
+These are actual [recommended extensions herein](.vscode/extensions.json).
+
+* Beside using [the extension][prettier], one may run `prettier` from the CLI:
+
+```bash
+# confirm formatting is ok:
+yarn format:check
+# correct the formatting:
+yarn format:write
+
+```
+
+## Status
+
+This extension is currently under [initial development][backlog].
```

Comment:
I think we can remove the status message. It's usable as is. WDYT?

Reply:
done